### ISSUE TITLE
DCH option 

### ISSUE THREAD

**Issue by Primeey** [fyr77/EnvyUpdate#16]
EnvyUpdate always leads you to download the standard version of the Nvidia driver but what if the user is required the DCH version? A option to let you chose between standard and DCH would be handy.

**Comment by fyr77** [comment]
Good call, I'll look into that when I have time.

**Comment by fyr77** [comment]
Interestingly, the API URL I used in EnvyUpdate defaults to the "normal" driver, but enetering the same settings on the website gives me the DCH driver. I need to figure out what causes this difference.

**Comment by fyr77** [comment]
Alright, the API uses `&dtcid=1` to specifiy a DCH driver. Not supplying this defaults to non-DCH (at least right now). `&dtcid=0` would force a non-DCH driver. Now to figure out how to detect DCH vs. non-DCH.

**Comment by fyr77** [comment]
I currently do not have access to a machine running Nvidia Graphics using the DCH driver. If anyone could allow me to poke around in their device manager via remote desktop or discord I could try to find out how to detect DCH vs non-DCH drivers. I expect a flag to be set somewhere...

Or, of course, if someone can tell me how to detet them or creates a pull request I could also finish implementing this.

In short: I need help or time to find a suitable system.

**Comment by Primeey** [comment]
I believe there's a DCH flag in the Windows Registry if that's any help? [I found this reddit post.](https://www.reddit.com/r/PowerShell/comments/cbjoz9/powershell_to_find_nvidia_dch_or_standard_driver/) and I can confirm on my system with the Nvidia DCH driver installed I have a DCHUVen key. Here's my full nvlddmkm registry folder, maybe you can compare it to a folder created when you install the standard driver 

![chrome_DUhN82B5ig](https://user-images.githubusercontent.com/8511642/90520982-e9465780-e161-11ea-9388-ac3bf7c782fb.png)

**Comment by fyr77** [comment]
@Primeey That's very helpful, thank you! I'll look into it.

**Comment by fyr77** [comment]
@Primeey Could you do me another favour and run the script located here: https://gist.github.com/fyr77/0ac16813c8f469d08789c9785f4368d2

If it works properly it should return "DCH found".

**Comment by Primeey** [comment]
Yep it works. Mind you I had to change my Execution Policy in PowerShell to run the unsigned script I don't know if that matters.

**Comment by fyr77** [comment]
@Primeey Thanks!

**Comment by fyr77** [comment]
Implemented in a42641f
